lib/ansible/modules/windows/win_firewall_rule.ps1

```diff
@@ -54,351 +72,390 @@ function preprocessAndCompare($key, $outputValue, $fwsettingValue) {
-            $rawoutput | Where {$_ -match '^([^:]+):\s*(\S.*)$'} | Foreach -Begin {
-                    $FirstRun = $true;
-                    $HashProps = @{};
+    try {
+        $command = "netsh advfirewall firewall show rule name=`"$($fwsettings.'Rule Name')`" verbose"
+        #$output = Get-NetFirewallRule -name $($fwsettings.'Rule Name')
+        $result.output = Invoke-Expression $command | Where { $_  }
+        $rc = $LASTEXITCODE
+        if ($rc -eq 1) {
+            $result.msg += @("No rule '$name' could be found")
+        } elseif ($rc -eq 0) {
+            # Process command output
+            $result.output | Where {$_ -match '^([^:]+):\s*(\S.*)$'} | ForEach -Begin {
+                    $FirstRun = $true
+                    $HashProps = @{}
                 } -Process {
-                    if (($Matches[1] -eq 'Rule Name') -and (!($FirstRun))) {
-                        #$output=New-Object -TypeName PSCustomObject -Property $HashProps;
-                        $output=$HashProps;
-                        $HashProps = @{};
-                    };
-                    $HashProps.$($Matches[1]) = $Matches[2];
-                    $FirstRun = $false;
+                    if (($Matches[1] -eq 'Rule Name') -and (-not $FirstRun)) {
+                        $output = $HashProps
+                        $HashProps = @{}
+                    }
+                    $HashProps.$($Matches[1]) = $Matches[2]
+                    $FirstRun = $false
                 } -End {
-                #$output=New-Object -TypeName PSCustomObject -Property $HashProps;
-                $output=$HashProps;
+                    $output = $HashProps
                 }
-        }
-        $exists=$false;
-        $correct=$true;
-        $diff=$false;
-        $multi=$false;
-        $correct=$false;
-        $difference=@();
-        $msg=@();
-        if ($($output|measure).count -gt 0) {
-            $exists=$true;
-            $msg += @("The rule '" + $fwsettings.'Rule Name' + "' exists.");
-            if ($($output|measure).count -gt 1) {
-                $multi=$true
-                $msg += @("The rule '" + $fwsettings.'Rule Name' + "' has multiple entries.");
-                ForEach($rule in $output.GetEnumerator()) {
+            if ($($output|measure).count -gt 0) {
+                $result.exists = $true
+                $result.msg += @("The rule '$($fwsettings.'Rule Name')' exists.")
+                if ($($output|measure).count -gt 1) {
+                    $result.multiple = $true
+                    $result.msg += @("The rule '$($fwsettings.'Rule Name')' has multiple entries.")
+                    if ($diff_support) {
+                        $result.diff.after = ConvertTo-SortedKV $fwsettings
+                        $result.diff.before = ConvertTo-SortedKV $rule $unsupported
+                    }
+                    ForEach($rule in $output.GetEnumerator()) {
```

Comment:
Well, I mostly translated the existing code to the newer structure, but now looking at what we are doing, it is no longer needed to do this. We can simply compare before and after to see if there is a difference.

I will fix this.